File: 백준/Silver/11052. 카드 구매하기/카드 구매하기.cs
using System;
namespace my
{
    class Program
    {
        static void Main()
        {
            int N = int.Parse(Console.ReadLine());
            string[] cards = Console.ReadLine().Split();
            int[] dp = new int[N+1];
            
            for (int i = 1; i <= N; i++)
            {
                for (int j = 1; j <= i; j++)
                {
                    dp[i] = Math.Max(dp[i], int.Parse(cards[j-1]) + dp[i-j]);
                }
            }
            Console.WriteLine(dp[N]);
        }
    }
}

File: 백준/Silver/11052. 카드 구매하기/카드 구매하기.py
from sys import stdin
input = stdin.readline

N = int(input())
cards = [0] + list(map(int, input().split()))
dp = [0 for _ in range(N+1)]

for i in range(1, N+1):
    for j in range(1, i+1):
        dp[i] = max(dp[i], cards[j] + dp[i-j])
print(dp[N])